File: ruby/lib/game_machine/remote_echo.rb
module GameMachine
  class RemoteEcho < GameActor
    
    def self.components
      ['EchoTest']
    end

    def on_receive(message)
      GameMachine.logger.debug("RemoteEcho client_id: #{message.get_client_connection.get_id} message: #{message}")
      ClientMessage.new(message,message.get_client_connection).send_to_client
    end
  end
end

File: server/java/server/src/main/java/io/gamemachine/core/RemoteEcho.java
package io.gamemachine.core;

import io.gamemachine.messages.Entity;
import akka.actor.UntypedActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;

public class RemoteEcho extends UntypedActor {

    public static String name = "GameMachine::GameSystems::RemoteEcho";
    LoggingAdapter logger = Logging.getLogger(getContext().system(), this);

    @Override
    public void onReceive(Object message) throws Exception {
        if (message instanceof Entity) {
            Entity entity = (Entity) message;
            PlayerMessage.tell(entity, entity.getPlayer().getId());
        }

    }

}
